src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -37,11 +37,13 @@ import Language.PureScript.TypeClassDictionaries
 import Language.PureScript.Types
 import qualified Language.PureScript.Constants as C
 
+type TypeClassDict =
+  TypeClassDictionaryInScope (Either Expr (Qualified Ident))
```

Comment:
I think I'd rather use a sum type here, with a constructor for named instances, and one for symbol dictionaries. Then instead of using `either id` in `mkDictionary`, you would pattern match, and include the instance generation code there. `isSymbolDictionary` would just apply the appropriate constructor.